src/test/check_getdns_context_set_dns_transport.h

```diff
@@ -71,108 +71,222 @@
+        GETDNS_RETURN_CONTEXT_UPDATE_FAIL, "Return code from getdns_context_set_dns_transport()");
+      ASSERT_RC(getdns_context_set_dns_transport_list(context, 2, transports),
+        GETDNS_RETURN_CONTEXT_UPDATE_FAIL, "Return code from getdns_context_set_dns_transport()");
+
+      CONTEXT_DESTROY;
+
+    }
+    END_TEST
+
+     START_TEST (getdns_context_set_dns_transport_stub_5)
      {
        /*
+       *  Request answer larger then 512 bytes but set UDP payload to that
        *  Call getdns_context_set_dns_transport() with value = GETDNS_TRANSPORT_UDP_ONLY
-       *  Define a callback routine for context changes and call getdns_context_set_context_update_callback() so that it gets called when there are context changes
-       *  getdns_context_set_resolution_type() to GETDNS_RESOLUTION_STUB
-       *  expect:  GETDNS_CONTEXT_CODE_RESOLUTION_TYPE
+       *  expect: Message uses UDP but is truncated
+       *  Call getdns_context_set_dns_transport() with value = GETDNS_TRANSPORT_TCP_ONLY
+       *  expect: Message uses TCP and is not truncated
+       *  Call getdns_context_set_dns_transport() with value = GETDNS_TRANSPORT_UDP_FIRST_AND_FALL_BACK_TO_TCP
+       *  expect: Message uses TCP and is not truncated
        */
        struct getdns_context *context = NULL;
        struct getdns_dict *response = NULL;
-       uint32_t ancount;
-       uint32_t arcount;
-       uint32_t nscount;
-       uint32_t tcp_ancount;
-       uint32_t tcp_arcount;
-       uint32_t tcp_nscount;
-       int udp_sum;
-       int tcp_sum;
-       
-        CONTEXT_CREATE(TRUE);
+       struct getdns_dict *extensions = getdns_dict_create();
+       uint32_t tc;
+       uint32_t transport;
+
+       /* Note that stricly this test just establishes that the requested transport
+          and the reported transport are consistent, it does not guarentee which
+          transport is used on the wire...*/
+
+       CONTEXT_CREATE(TRUE);
 
+       ASSERT_RC(getdns_context_set_resolution_type(context, GETDNS_RESOLUTION_STUB),
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_resolution_type()");
+       ASSERT_RC(getdns_dict_set_int(extensions,"return_call_reporting", GETDNS_EXTENSION_TRUE),
+         GETDNS_RETURN_GOOD, "Return code from getdns_dict_set_int()");
 
+       /* Request a response that should be truncated over UDP */
        ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_UDP_ONLY),
-         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
-     
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");
+       ASSERT_RC(getdns_context_set_edns_maximum_udp_payload_size(context, 512),
+           GETDNS_RETURN_GOOD, "Return code from getdns_context_set_edns_maximum_udp_payload_size()"); 
+       ASSERT_RC(getdns_context_set_edns_do_bit(context, 1),
+           GETDNS_RETURN_GOOD, "Return code from getdns_context_set_edns_do_bit()");
```

Comment:
This test fails on travis, because the upstream there doesn't provide signatures with the DNSKEY set.  The test should ask either for a something that exceeds 512 bytes even without signatures, or set the upstream to a resolver that does return signatures.

_443._tcp.full.getdnsapi.net TLSA will be larger than 512 bytes even without signatures, but perhaps it would be better to put a custom TXT rrset in the zone for the purpose of this test.